src/cogent3/core/new_alignment.py

```diff
@@ -3449,8 +3601,14 @@ def __repr__(self):
         return f"{len(self.names)} x {len(self)} {self.moltype.label} alignment: {seqs}"
 
     def __len__(self):
+        # refactor: design
+        # should the len of a sliced alignment be updated? i.e.,
+        # return len(self._slice_record)
```

Comment:
this is the correct line, any code that fails as a result of this change is wrong

Reply:
will push through in the next PR